tests/unit/test_collector.py

```diff
@@ -189,10 +191,43 @@ def test_get_machine_type(self, collector_daemon, mac_address, expect_machine_ty
                 }
             }
         }
-        machine_type = statsd._get_machine_type(machine)
+        machine_type = statsd._get_machine_type(machine, "dummy-0")
 
         assert machine_type.value == expect_machine_type
 
+    def test_get_machine_type_interface_skip(self, collector_daemon):
+        """Test that blacklisted interfaces are not used to detect machine type."""
+        statsd = collector_daemon()
+        skip_interfaces = ["virb*", "tap-*"]
```

Comment:
these will match anyway, but you probably meant `virb.*` and `tap-.*`

Reply:
Come to think of it, `["^virbr", "^tap-"]` are probably good enough, right?

Reply:
they will all match, even just `["virbr", "tap"]`; it depends on how specific you want them to be. It's just that `virb*` and `tap-*` look like typos and not what you meant